core/trino-main/src/main/java/io/trino/operator/aggregation/TypedSet.java

```diff
@@ -223,7 +223,9 @@ public void add(Block block, int position)
         int hashPosition = getHashPositionOfElement(block, position);
         if (blockPositionByHash.getInt(hashPosition) == EMPTY_SLOT) {
             addNewElement(hashPosition, block, position);
+            return true;
         }
+        return false;
```

Comment:
do you know why we don't exist early after `containsNullElement = true;` above?

add unit test coverage for adding null element for the first and second time

Reply:
We don't exit early because we still have to add the null if it wasn't already there. We could include a check if null is already contained and exit early, though.

([`TestTypedSet.testGetElementPositionWithNull`](https://github.com/trinodb/trino/blob/fe608f2723842037ff620d612a706900e79c52c8/core/trino-main/src/test/java/io/trino/operator/aggregation/TestTypedSet.java#L83) tests adding multiple nulls.)